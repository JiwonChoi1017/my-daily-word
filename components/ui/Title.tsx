import React from "react";
import classes from "@/styles/Title.module.css";

/** Props. */
interface Props {
  /** タイトル. */
  title: string;
  /** (任意)サブタイトル. */
  subtitle?: string;
}

/**
 * タイトル.
 *
 * @param {Props} props.
 * @returns {JSX.Element} タイトル.
 */
const Title = ({ title, subtitle }: Props) => {
  return (
    <div className={classes.title__wrap}>
      <span className={classes.title}>{title}</span>
      {subtitle && <span className={classes.title}>{subtitle}</span>}
    </div>
  );
};

export default Title;
