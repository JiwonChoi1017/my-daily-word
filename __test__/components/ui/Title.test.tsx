import React from "react";
import Title from "@/components/ui/Title";
import { render } from "@testing-library/react";

describe("Titleのテスト", () => {
  test("Title: スナップショットテスト", () => {
    const { container } = render(<Title title="Title1" subtitle="SubTitle1" />);

    expect(container).toMatchSnapshot();
  });
});
