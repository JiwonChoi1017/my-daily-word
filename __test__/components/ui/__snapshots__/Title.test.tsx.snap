// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titleのテスト Title: スナップショットテスト 1`] = `
<div>
  <div
    class="title__wrap"
  >
    <span
      class="title"
    >
      Title1
    </span>
    <span
      class="title"
    >
      SubTitle1
    </span>
  </div>
</div>
`;
